improver/calibration/rainforest_calibration.py

```diff
@@ -461,124 +465,123 @@ def _calculate_error_probabilities(
                 Cubelist containing the independent feature variables for prediction.
 
         Returns:
-            A cube containing error exceedence probabilities.
+            A cube containing threshold exceedence probabilities.
 
         Raises:
             ValueError:
                 If an unsupported model object is passed. Expects lightgbm Booster, or
                 treelite_runtime Predictor (if treelite dependency is available).
         """
 
-        error_probability_cube = self._prepare_error_probability_cube(forecast_cube)
+        threshold_probability_cube = self._prepare_threshold_probability_cube(
+            forecast_cube
+        )
 
         input_dataset = self._prepare_features_array(feature_cubes)
 
-        forecast_data = forecast_cube.data.ravel()
+        lead_time_hours = forecast_cube.coord("forecast_period").points[0] / (
+            SECONDS_IN_MINUTE * MINUTES_IN_HOUR
+        )
 
         self._evaluate_probabilities(
-            forecast_data,
-            input_dataset,
-            forecast_cube.name(),
-            forecast_cube.units,
-            error_probability_cube.data,
+            input_dataset, lead_time_hours, threshold_probability_cube.data,
         )
 
         # Enforcing monotonicity
-        error_probability_cube.data = self._make_decreasing(error_probability_cube.data)
+        threshold_probability_cube.data = self._make_decreasing(
+            threshold_probability_cube.data
+        )
 
-        return error_probability_cube
+        return threshold_probability_cube
 
     def _get_ensemble_distributions(
```

Comment:
it's similar, but we're interpolating between different sets of thresholds, rather than different sets of percentiles.

Reply:
No worries. This is out of scope to move this functionality, but would it be worth adding a comment to flag that this could be moved to `ensemble_colpula_coupling` if a `ResampleThresholds` functionality was needed more generally?

Reply:
Happy to leave this thread as this suggestion is non-critical.